caas/kubernetes/provider/k8s.go

```diff
@@ -1319,7 +1319,19 @@ func makeUnitSpec(appName string, podSpec *caas.PodSpec) (*unitSpec, error) {
 		if spec.ReadinessProbe != nil {
 			unitSpec.Pod.Containers[i].ReadinessProbe = spec.ReadinessProbe
 		}
+
+		if c.Constraints != nil {
```

Comment:
This should go in a separate configureDevices() func like we do for configureStorage()